### PR TITLE
To many elements to index

### PR DESCRIPTION
**Pull request by tomekstach**
I don't know why in the query `#__tags` is twice. Once as a `a` and second time as a `b`. For me it works perfectly when we remove `b`. I had a problem with this - for 1100 tags on my websites, Smart Search Indexer found more than 1233000 items to index.

Pull Request for Issue #13866.

### Summary of Changes

Remove `#__tags AS b` from the query and change the `b.created_user_id` to the `a.created_user_id` in the JOIN part of the query.

### Testing Instructions

Use `php cli/finder_indexer.php` to check if the number of tags is the same as a number of items to index.

### Expected result

For the number of tags have the same number of items for index.

### Actual result

For 1100 tags, I have more than 1233000 items to index.

### Documentation Changes Required

### CODE REVIEW

File: plugins/finder/tags/tags.php [hunk 1]
```diff
@@ -313,8 +313,7 @@ protected function getListQuery($query = null)
 
 		// Join the #__users table
 		$query->select('u.name AS author')
-			->join('LEFT', '#__users AS u ON u.id = b.created_user_id')
-			->from('#__tags AS b');
+			->join('LEFT', '#__users AS u ON u.id = b.created_user_id');
```
Comment (dgt41) [inline plugins/finder/tags/tags.php]: a.created_user_id??

Comment (tomekstach) [inline plugins/finder/tags/tags.php]: Yes, you are right - it is my mistake.

**Comment by Hackwar** [comment]
I can confirm this issue and tested this item successfully.

**Comment by Hackwar** [comment]
I have tested this item :white_check_mark: successfully on ac3aa1216cd9121ea121b102440b38f42c1d873b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13868">issues.joomla.org/tracker/joomla-cms/13868</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ac3aa1216cd9121ea121b102440b38f42c1d873b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13868">issues.joomla.org/tracker/joomla-cms/13868</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13868">issues.joomla.org/tracker/joomla-cms/13868</a>.</sub>

**Comment by ggppdk** [comment]
I have tested this item :white_check_mark: successfully on ac3aa1216cd9121ea121b102440b38f42c1d873b<br /><br />Tag indexing works properly 
and

before PR we get `FROM #__tags AS a, #__tags AS b`
a million rows to index for 1000 tags

```sql
SELECT a.id, ...
	FROM #__tags AS a, #__tags AS b
	LEFT JOIN #__users AS u ON u.id = b.created_user_id
	WHERE `a`.`id` > 1
```

After PR `FROM #__tags AS a`
1000 rows to index for 1000 tags

```sql	
SELECT a.id, ...
	FROM #__tags AS a
	LEFT JOIN #__users AS u ON u.id = a.created_user_id
	WHERE `a`.`id` > 1
```
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13868">issues.joomla.org/tracker/joomla-cms/13868</a>.</sub>

**Pull request merged**